guacamole-common-js/src/main/webapp/modules/OnScreenKeyboard.js

```diff
@@ -455,6 +455,17 @@ Guacamole.OnScreenKeyboard = function(layout) {
 
     };
 
+    /**
+     * Resets the state of this keyboard, releasing all keys, and firing keyup
+     * events for each released key.
+     */
+    this.reset = function() {
+        for (var keysym in pressed) {
```

Comment:
The `pressed` structure within `Guacamole.OnScreenKeyboard` is indexed by key name, not keysym:

https://github.com/apache/guacamole-client/blob/5db2e3cae75a658b33515112a39eb2f5f8eef4a8/guacamole-common-js/src/main/webapp/modules/OnScreenKeyboard.js#L51-L59